sys/compat/freebsd64/freebsd64_jail.c

```diff
@@ -98,7 +101,7 @@ freebsd64_jail(struct thread *td, struct freebsd64_jail_args *uap)
 	case 2:	{ /* JAIL_API_VERSION */
 		struct jail64 j;
```

Comment:
`struct jail64` still contains `char *`, we should move them to `uint64_t` otherwise they will break with the purecap kernel.

Reply:
These have commented out __ptr64 allocations already.  Not sure if those work yet, but if they do we might prefer those?  @brooksdavis might have thoughts on that.  We can always resolve these in a future change as the code in the current commit works either way.

Reply:
I do agree that `__ptr64` is the correct way of handling this long term. It was my understanding that it is not yet ready though. If it works fine in the hybrid case we can leave them for now.

Reply:
I'd eventually like to move to __ptr64 or a struct hack (I think @qwattash has at least prototyped that), but I'm fine either way at this juncture.

Reply:
I'm fine with keeping this as is or switching to int64_t. Probably in the short term switching to int64_t is best.

Longer term it's a little unclear what we want.  I like __ptr64 because I hate discarding type information, but I also want it to be non-dereferencible so programmer errors are caught on non-CHERI architectures.

Reply:
I've added a change to switch to uint64_t.